src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImpl.java

```diff
@@ -45,24 +47,55 @@
 public class BitbucketWebhookTriggerImpl extends Trigger<Job<?, ?>>
         implements BitbucketWebhookTrigger {
 
+    //the version (of this class) where the PR trigger was introduced. Version is 0 based.
+    private static final int BUILD_ON_PULL_REQUEST_VERSION = 1;
     private static final Logger LOGGER = Logger.getLogger(BitbucketWebhookTriggerImpl.class.getName());
 
+    private boolean pullRequestTrigger;
+    private boolean refTrigger;
+    /**
+     * This exists as a simple upgrade task. Old classes will de-serialise this to default value (of 0). New
+     * classes will serialise the actual value that was stored. Because the constructor is not run during de-serialisation
+     * we can safely set the value in the constructor to indicate which version this class is.
+     */
+    private final int version;
+
     @SuppressWarnings("RedundantNoArgConstructor") // Required for Stapler
     @DataBoundConstructor
-    public BitbucketWebhookTriggerImpl() {
+    public BitbucketWebhookTriggerImpl(boolean pullRequestTrigger, boolean refTrigger) {
+        version = BUILD_ON_PULL_REQUEST_VERSION;
+        this.pullRequestTrigger = pullRequestTrigger;
+        this.refTrigger = refTrigger;
     }
 
     @Override
     public BitbucketWebhookTriggerDescriptor getDescriptor() {
         return (BitbucketWebhookTriggerDescriptor) super.getDescriptor();
     }
 
+    public boolean isPullRequestTrigger() {
+        return pullRequestTrigger;
+    }
+
+    public boolean isRefTrigger() {
+        /*
+         * If it is an old version we should return true as that is the old default.
+         * If it is a new version we should use the value that was set.
+         */
+        if (version < BUILD_ON_PULL_REQUEST_VERSION) {
+            return true;
+        } else {
+            return refTrigger;
+        }
+    }
+
     @Override
-    public void trigger(BitbucketWebhookTriggerRequest triggerRequest) {
-        SCMTriggerItem triggerItem = asSCMTriggerItem(job);
-        if (triggerItem != null) {
-            getDescriptor().schedule(job, triggerItem, triggerRequest);
+    public boolean isApplicableForEventType(AbstractWebhookEvent event) {
+        if (event instanceof PullRequestWebhookEvent && isPullRequestTrigger()) {
```

Comment:
this is unfortunate due to the generic `PullRequestWebhookEvent` so we do this now, and the plan is to change the event hierarchy such that we will actually have `PROpened` and `PRDeclined` type events. I wanted to keep this PR as small as possible so that will come in a separate PR

Reply:
it matters, due to the upgrade task, so we must *always* use the getter. 

And yes, we can do more fancy things relatively easy now. It is quite nice, have a look at my [other PR](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/249/files#diff-99690be9bbe5b58c1967fc9835482fafa0d121eae2ea01460f2f3c8e0ad9b36b)